akd_core/src/proto/mod.rs

```diff
@@ -99,11 +99,33 @@ macro_rules! convert_from_vector {
 // NodeLabel
 // ==============================================================
 
+fn minimize_label_bytes(v: &[u8; 32]) -> Vec<u8> {
+    let mut i = 31;
+    for byte in v.iter().rev() {
+        if *byte != 0u8 {
+            break;
+        }
+
+        if i > 0 {
+            i -= 1;
+        } else {
+            return Vec::new();
+        }
+    }
+    v[0..=i].to_vec()
+}
+
+fn parse_min_label(v: &[u8]) -> [u8; 32] {
+    let mut out = [0u8; 32];
+    out[..v.len()].copy_from_slice(v);
+    out
+}
+
 impl From<&crate::NodeLabel> for specs::types::NodeLabel {
     fn from(input: &crate::NodeLabel) -> Self {
         Self {
             label_len: Some(input.label_len),
-            label_val: Some(input.label_val.to_vec()),
+            label_val: Some(minimize_label_bytes(&input.label_val)),
```

Comment:
Rather than iterating and searching for the the beginning of the trailing zeros, we can just use the label_len parameter here. In other words, just take the slice of `&input.label_val[..input.label_len]` and that should work!